Implement beginning point for server and database creation

Ciaranengelbrecht opened the pull request:
Have restructured the repo, introduced Flask, SQLAlchemy to enable user authentication, so far seems to be working relatively well with mild testing of account creation, must install the requirements.txt file and also setup a secret key .env file on your local system (currently is literally just set to: 'SECRET_KEY=maythe4thbewithyou' in the .env file at the root of the repo).
User is able to create account or sign in if the account has previously been created and redirects to main page.
Still yet to implement WTForms, I attempted it but this broke a few things and the login window was no longer popping up.
Yet to fully implement logging out functionality, will need to add the html/css for this on the notes.html page.
Still working on this so won't be merging until working better according to the project specs and everyone is happy with it.

darkksuns reviewed (approved):
This looks so good! Happy for you to merge this.
One small thing is that when opening the sign up and log in pop-ups the page puts a scrollbar on the side when theres no content to scroll.

Ciaranengelbrecht commented:
Thanks for the suggestion I have now removed the scrollable area when the authentication windows open.
I have also implemented the logout functionality so users will be redirected to the auth page when not currently in an active and authenticated session

Ciaranengelbrecht commented:
Screenshots of current landing page with full functionality

<img width="484" alt="Screenshot 2024-05-09 at 12 57 48 PM" src="https://github.com/DJRyLau/CITS3403-Project/assets/131735166/21b15c94-4e3d-47dc-9c8f-77ca64dcf6a7">

<img width="1440" alt="Screenshot 2024-05-09 at 12 57 09 PM" src="https://github.com/DJRyLau/CITS3403-Project/assets/131735166/ab378b53-e3c9-485f-9230-7c0f81c795d6">

<img width="1440" alt="Screenshot 2024-05-09 at 12 57 22 PM" src="https://github.com/DJRyLau/CITS3403-Project/assets/131735166/6252ddc8-2fa3-43bc-860a-98715c6960a8">
